File: examples/list_namespaces.rs
use objectscale_client::client::ManagementClient;

fn main() {
    let endpoint = "https://10.225.108.217:4443";
    let username = "root";
    let password = "Password123!";
    let insecure = true;

    let name_prefix = "";

    let mut client =
        ManagementClient::new(endpoint, username, password, insecure).expect("management client");
    let namespaces = client
        .list_namespaces(name_prefix)
        .expect("list namespaces");
    println!("List namespaces: {:?}", namespaces);
}

File: python/examples/list_namespaces.py
import objectscale_client

def main():
    endpoint = "https://10.225.108.217:4443"
    username = "root"
    password = "Password123!"
    insecure = True

    try:
        client = objectscale_client.client.ManagementClient(endpoint, username, password, insecure)
        namespaces = client.list_namespaces("")
        for namespace in namespaces:
            print(namespace)
    except Exception as e:
        print("Failed to list namespaces:", e)


if __name__ == '__main__':
    main()
